cachito/workers/scm.py

```diff
@@ -0,0 +1,201 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+import os
+from abc import ABC, abstractmethod
+import urllib.parse
+import shutil
+import tarfile
+import tempfile
+import subprocess
+
+import requests
+
+from cachito.errors import ConfigError, CachitoError
+
+
+log = logging.getLogger(__name__)
+
+
+# TODO: Write tests
+class SCM(ABC):
+    """The base class for interacting with source control."""
+
+    def __init__(self, url, ref):
+        """
+        Initialize the SCM class.
+
+        :param str url: the source control URL to the repository to fetch
+        :param str ref: the source control reference
+        """
+        super().__init__()
+        self.url = url
+        self.ref = ref
+        self._archives_dir = None
+        self._archive_path = None
+        self._repo_name = None
+        # Validate that the base directory can be retrieved
+        self.archives_dir
+
+    @property
+    def archive_path(self):
+        """
+        Get the path to where the archive for a particular SCM reference should be.
+
+        :return: the path to the archive
+        :rtype: str
+        """
+        if not self._archive_path:
+            directory = os.path.join(self.archives_dir, *self.repo_name.split('/'))
```

Comment:
Is splitting necessary? Won't this produce the same thing? 
```python
os.path.join(self.archives_dir, self.repo_name)
```

Reply:
It would on systems that use forward slashes in their paths. I tend to not assume that when writing code, but it's probably safe to assume we won't be supporting Windows :stuck_out_tongue:

Reply:
Fair point :)